wallets/core/src/namespaces/evm/actions.ts

```diff
@@ -95,3 +95,35 @@ export function changeAccountSubscriber(
     },
   ];
 }
+
+export function changeChainSubscriber(
+  instance: () => ProviderAPI
+): [Subscriber<EvmActions>, SubscriberCleanUp<EvmActions>] {
+  let eventCallback: EIP1193EventMap['chainChanged'];
+
+  return [
+    (context) => {
+      const evmInstance = instance();
+
+      if (!evmInstance) {
+        throw new Error(
+          'Trying to subscribe to your EVM wallet, but seems its instance is not available.'
+        );
+      }
+
+      const [, setState] = context.state();
+
+      eventCallback = async (chainId: string) => {
+        setState('network', chainId);
```

Comment:
@yeager-eren 

Is it correct here that I am updating the network by `chainId`? Or should I find the network by `chainId`, and update the network?